src/Microsoft.DotNet.ProjectJsonMigration/Rules/AddDefaultsToProjectRule.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using Microsoft.DotNet.ProjectJsonMigration.Transforms;
+using Microsoft.Build.Construction;
+using System.Collections.Generic;
+using Microsoft.DotNet.ProjectJsonMigration.Models;
+using System;
+using System.IO;
+using System.Reflection;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.ProjectJsonMigration.Rules
+{
+    public class AddDefaultsToProjectRule : IMigrationRule
+    {
+        internal const string c_DefaultsProjectElementContainerLabel = "MigrationDefaultsTempContainer";
+        internal const string c_SdkDefaultsJsonFileName = "sdkdefaults.json";
+
+        private readonly ITransformApplicator _transformApplicator;
+
+        public AddDefaultsToProjectRule(ITransformApplicator transformApplicator = null)
+        {
+            _transformApplicator = transformApplicator ?? new TransformApplicator();
+        }
+
+        public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migrationRuleInputs)
+        {
+            SerializableMigrationDefaultsInfo defaults = ResolveDefaults();
+
+            var project = migrationRuleInputs.OutputMSBuildProject;
+            var defaultsPropertyGroups = new Dictionary<string, ProjectPropertyGroupElement>();
+            var defaultsItemGroups = new Dictionary<string, ProjectItemGroupElement>();
+
+            AddDefaultPropertiesToProject(defaults, project, defaultsPropertyGroups);
+            AddDefaultItemsToProject(defaults, project, defaultsItemGroups);
+        }
+
+        private void AddDefaultItemsToProject(
+            SerializableMigrationDefaultsInfo defaults, 
+            ProjectRootElement project, 
+            Dictionary<string, ProjectItemGroupElement> defaultsItemGroups)
+        {
+            foreach (var itemInfo in defaults.Items)
+            {
+                ProjectItemGroupElement itemGroup;
+                var parentCondition = itemInfo.ParentCondition ?? string.Empty;
+
+                if (!defaultsItemGroups.TryGetValue(parentCondition, out itemGroup))
+                {
+                    itemGroup = project.AddItemGroup();
```

Comment:
We don't have any rules (or plan on having any rules in the future) that will add things to these groups, right?

Because if we add these groups here, and then some other rule adds to this group, and then we delete the 'defaults' group at the end of migration, we will lose that other rules' changes.